R/guide-legend.r

```diff
@@ -249,7 +256,18 @@ guide_geom.legend <- function(guide, layers, default_mapping) {
 
     if (length(matched) > 0) {
       # This layer contributes to the legend
-      if (is.na(layer$show.legend) || layer$show.legend) {
+
+      # rename color to colour
+      layer$show.legend <- unlist(rename_aes(as.list(layer$show.legend)))
```

Comment:
I think this rename would be better off inside the branch where names are found.  

Also why are you converting to a list and back?